app/core/pokemon-entity.ts

```diff
@@ -444,7 +444,11 @@ export class PokemonEntity extends Schema implements IPokemonEntity {
       target.hp = min(1)(target.hp + value)
     }
     update(this)
-    this.life = max(this.hp)(this.life + value)
+    if (value > 0) {
+      this.life = max(this.hp)(this.life + value)
+    } else {
+      this.life = max(this.hp)(this.life)
```

Comment:
this change doesn't make sense to me, if a unit lose max HP it should lose life as well